LibraBFT/ImplShared/Util/Dijkstra/EitherD.agda

```diff
@@ -107,3 +110,50 @@ EitherD-contract (EitherD-maybe nothing f₁ f₂) P wp =
   EitherD-contract f₁ P (proj₁ wp refl)
 EitherD-contract (EitherD-maybe (just x) f₁ f₂) P wp =
   EitherD-contract (f₂ x) P (proj₂ wp x refl)
+
+EitherD-⇒
```

Comment:
This seems to be the center of the PR - adding (and using) weakest-pre machinery to `Either`.
I thought it already existed.

Reply:
Most of the machinery already exists.  Here we are just extending it a bit to support the proofs in the PR.